lib/reek/cli/silencer.rb

```diff
@@ -3,11 +3,24 @@ module CLI
     # CLI silencer
     # @api private
     module Silencer
-      def self.silently
-        old_verbose, $VERBOSE = $VERBOSE, nil
+      def self.silently(stdout: false, stderr: false)
+        old_verbose = $VERBOSE
+        $VERBOSE = nil
+        if stdout
+          old_stdout = $stdout
+          $stdout = StringIO.new
+        end
+
+        if stderr
+          old_stderr = $stderr
+          $stderr = StringIO.new
+        end
+
         yield if block_given?
       ensure
         $VERBOSE = old_verbose
+        $stderr = old_stderr if stderr
+        $stdout = old_stdout if stdout
```

Comment:
>>  I’m thinking whether reassigning to STDERR and STDOUT wouldn’t be more canonical (and significantly simpler). They’re supposed to hold the references to the canonical streams, and the only reason for caching old_stderr/old_stdout by hand would be if this silencer was itself embedded in another piece of code tha alters $stderr/$stdout.

Ah, I like that.:)

>>  how about making the kwargs the target streams (so the caller would pass StringIO.new or whatever it needs rather than true)?

But so far in 100% of all cases where we use "silently" we just want to, uhm, silence the corresponding stream and we don't care if that is a StringIO, a "true" or a unicorn on a rainbow.
So forcing the caller to pass in something else we generate the impression that this is important, when in fact it isn't. 
So I agree with all of your other suggestions, but I'd rather keep the flags.